python-backend/app/api/mines/mine/resources/mine.py

```diff
@@ -23,114 +23,119 @@
+from app.api.mines.mine_api_models import MINE_LIST_MODEL, MINE
 # FIXME: Model import from outside of its namespace
 # This breaks micro-service architecture and is done
 # for search performance until search can be refactored
 from ....permits.permit.models.permit import Permit
 
 
-class MineResource(Resource, UserMixin, ErrorMixin):
+class MineListResource(Resource, UserMixin):
     parser = reqparse.RequestParser()
     parser.add_argument(
-        'mine_name', type=str, help='Name of the mine.', trim=True, store_missing=False)
+        'mine_name', type=str, help='Name of the mine.', trim=True, required=True, location='json')
     parser.add_argument(
         'mine_note',
         type=str,
         help='Any additional notes to be added to the mine.',
         trim=True,
-        store_missing=False)
-    parser.add_argument(
-        'tenure_number_id',
-        type=int,
-        help='Tenure number for the mine.',
-        trim=True,
-        store_missing=False)
+        store_missing=False,
+        location='json')
     parser.add_argument(
         'longitude',
         type=lambda x: Decimal(x) if x else None,
         help='Longitude point for the mine.',
-        store_missing=False)
+        store_missing=False,
+        location='json')
     parser.add_argument(
         'latitude',
         type=lambda x: Decimal(x) if x else None,
         help='Latitude point for the mine.',
-        store_missing=False)
+        store_missing=False,
+        location='json')
     parser.add_argument(
         'mine_status',
         action='split',
         help=
         'Status of the mine, to be given as a comma separated string value. Ex: status_code, status_reason_code, status_sub_reason_code ',
-        store_missing=False)
+        required=True,
+        location='json')
     parser.add_argument(
         'major_mine_ind',
         type=inputs.boolean,
         help='Indication if mine is major_mine_ind or regional. Accepts "true", "false", "1", "0".',
-        store_missing=False)
+        store_missing=False,
+        location='json')
     parser.add_argument(
-        'mine_region', type=str, help='Region for the mine.', trim=True, store_missing=False)
+        'mine_region',
+        type=str,
+        help='Region for the mine.',
+        trim=True,
+        required=True,
+        location='json')
 
     @api.doc(
         params={
-            'mine_no_or_guid':
-            'Mine number or guid. If not provided a paginated list of mines will be returned.'
-        })
+            'per_page': 'The number of results to be returned per page.',
+            'page': 'the current page number to be displayed.',
+            'search': 'The search term.',
+            'commodity': 'A commodity to filter the mine list by.',
+            'status': 'A mine status to filter the mine list by.',
+            'tenure': 'A mine tenure type to filter the mine list by.',
+            'region': 'A mine region to filter the mine list by.',
```

Comment:
The grammar nerd in me is really bothered by the prepositions at the end of these statements